app/api/salesforce/messages/route.ts

```diff
@@ -38,7 +40,23 @@ async function handleMessageStreaming(
   const stream = new ReadableStream({
     async start(controller) {
       let ack = -1;
+      let keepaliveInterval: NodeJS.Timeout | undefined;
+
+      const resetKeepalive = () => {
+        if (keepaliveInterval) {
+          clearInterval(keepaliveInterval);
+        }
+        keepaliveInterval = setInterval(() => {
+          if (!req.signal.aborted) {
+            controller.enqueue(new TextEncoder().encode(":\n\n")); // Send minimal keepalive message
```

Comment:
the `:` is a little cryptic, perhaps we can conform to the [SSE format](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event) and instead send  
```suggestion
         controller.enqueue(new TextEncoder().encode("event: keepalive\ndata: {}\n\n"));
```